### ISSUE TITLE
[Enhancement] All releases should be signed

### ISSUE THREAD

**Issue by c4f3a0ce** [ProtonVPN/protonvpn-cli-ng#72]
Since the package requires significant privileges it is only reasonable to expect that all releases  will be signed, preferably using a key under control of ProtonVPN and easily verifiable.

**Comment by Rafficer** [comment]
I don't think pip supports signatures and verification, does it?

**Comment by c4f3a0ce** [comment]
That's correct. `pip` doesn't provide any support for signature verification, though PyPi does support signature upload and retrieval, so it is possible to verify signature manually.

In this case I was thinking more about signing release tags - some of the releases (v.2.0.0, v2.1.0), where already signed (thanks), but it would be nice for this to be a norm. It doesn't add much overhead, but provides a bit more confidence.

**Comment by Rafficer** [comment]
No releases are signed. The releases where you see the "verified" tag are just releases where the last commit is signed. All of my commits are signed if you check the PRs, but as soon as they get merged they aren't signed anymore (cause they are new commits then).

v2.0.0 and v2.1.0 have signed commit because I still had direct access to the master branch back then.

**Comment by Rafficer** [comment]
But I guess it's best if @kaplun can say something here. I can't decide this because I don't have access to pypi anyway.

**Comment by kaplun** [comment]
Sorry, seeing this only now. Yes. This is a good point. I'll investigate how to best do this :+1:

**Comment by jiblime** [comment]
Is pip necessary?

**Comment by jwflory** [comment]
I am working on creating an official package for Fedora and EPEL. From a packager point-of-view, it is possible to verify GPG signatures during RPM package build time. If a signature was attached to each release, I could integrate a GPG verification into the RPM package spec.

**Comment by exprez135** [comment]
PGP signatures would be amazing with this package. With the Arch AUR, where we install without using pip, we can also configure the PKGBUILD to automatically verify signatures.

**Comment by jwflory** [comment]
v2.2.4 is packaged in Fedora and EPEL already. v2.2.6 is on the way soon. The Fedora packages are signed with PGP signatures using Fedora's official package keys, verifying that the package is built from the original source code blobs cloned from this GitHub repository.

Some progress to involve ProtonVPN in Fedora package administration along with me is here: https://bugzilla.redhat.com/show_bug.cgi?id=1929202
